lib/spack/env/cc

```diff
@@ -320,7 +320,7 @@ esac
 #
 # Unset pesky environment variables that could affect build sanity.
 #
-unset LD_LIBRARY_PATH
+#unset LD_LIBRARY_PATH
```

Comment:
i think this does not belong to this PR and requires a separate discussion. @alalazo @tgamblin @adamjstewart ping.

I suggest creating a separate PR with this change only, or at least remove it from this PR.